tests/RLS/TableManagerTest.php

```diff
@@ -766,13 +710,91 @@
         $table->foreignId('highlighted_comment_id')->nullable()->comment('no-rls')->constrained('comments');
     });
 
+    // Add a foreign key constraint to the comments table to introduce a recursive relationship
+    // Note that the comments table still has the post_id foreign key that leads to the tenants table
     Schema::table('comments', function (Blueprint $table) {
         $table->foreignId('recursive_post_id')->comment('rls')->constrained('recursive_posts');
     });
 
-    expect(fn () => app(TableRLSManager::class)->generateTrees())->not()->toThrow(RecursiveRelationshipException::class);
+    // No exception thrown because
+    // the highlighted_comment_id foreign key has a no-rls comment
+    $shortestPaths = app(TableRLSManager::class)->shortestPaths();
+
+    expect(array_keys($shortestPaths))
+        ->toContain('posts', 'comments')
+        // Shortest paths do not include the recursive_posts table
+        // because it has a 'no-rls' comment on its only foreign key
+        ->not()->toContain('recursive_posts');
 });
 
+test('table manager can generate paths leading through comment constraint columns', function() {
```

Comment:
```suggestion
test('table manager can generate paths leading through columns with comment constraint', function() {
```

Imagine we have `$table->string('tenant_id')->comment('rls tenants.id')` in a migration. `tenant_id` is a column with a comment constraint. "column with comment constraint" sounds verbose, "comment constraint column" sounds weird/vague (and still pretty verbose?)

I'd probably go with the former for clarity here.. What would you prefer @stancl?